File: problems/01_arrays_and_hashing/0242_valid_anagram/solution.swift
// Problem: 242. Valid Anagram
// Difficulty: Easy
// URL: https://leetcode.com/problems/valid-anagram/
//
// 【解法演進】排序比較 O(n log n) → 計數 O(n)（最佳）
// 【最佳解核心思路】長度不同直接 false；s +1、t -1，最後檢查全 0。
// 【複雜度】時間：O(n)  空間：O(1)

import Foundation

class Solution {
    func isAnagram(_ s: String, _ t: String) -> Bool {
        // TODO: implement
        return false
    }

    // 暴力解：排序比較
    func isAnagram_brute(_ s: String, _ t: String) -> Bool {
        // TODO: implement
        return false
    }
}

func checkBool(_ actual: Bool, _ expected: Bool, _ name: String) {
    if actual == expected { print("✓ \(name)") }
    else { print("✗ \(name): expected \(expected) but got \(actual)") }
}

// MARK: - Tests
// let s = Solution()

// Test 1
// let result = s.solve(...)

print("All tests passed!")

File: problems/01_arrays_and_hashing/0242_valid_anagram/solution.py
"""
Problem: 242. Valid Anagram
Difficulty: Easy
URL: https://leetcode.com/problems/valid-anagram/

【解法演進】排序比較 O(n log n) → 計數陣列/字典 O(n)（最佳）
【最佳解核心思路】
    s 計數 +1、t 計數 -1，最後檢查全為 0；長度不同直接 False。
【複雜度】時間：O(n)  空間：O(1)（固定 26 槽）；Unicode 則 O(k)
"""

from collections import Counter

class Solution:
    # 最佳解：計數陣列
    def isAnagram(self, s: str, t: str) -> bool:
        pass

    def isAnagram_counter(self, s: str, t: str) -> bool:
        pass

    def isAnagram_brute(self, s: str, t: str) -> bool:
        pass


if __name__ == "__main__":
    solution = Solution()

    # Test 1
    # result = solution.solve(...)
    # assert result == expected

    print("All tests passed!")
